dask/tests/test_optimization.py

```diff
@@ -1116,9 +1116,18 @@ def test_SubgraphCallable():
     }
 
     f = SubgraphCallable(dsk, "h", ["in1", "in2"], name="test")
+
     assert f.name == "test"
     assert repr(f) == "test"
 
+    f2 = SubgraphCallable(dsk, "h", ["in1", "in2"], name="test")
+    assert f == f2
+
+    f3 = SubgraphCallable(dsk, "g", ["in1", "in2"], name="test")
+    assert f != f3
+
+    dict(f=None)
+
```

Comment:
```suggestion
    hash(SubgraphCallable(None, None, [None]))
    assert hash(f3) != hash(f2)
```

Maybe add a test `hash(SubgraphCallable(None, None, [None]))` does not raise ? I think that if hash raise they might just get ignored, or fallback on something else.